File: viasoft.licensing.licensingmanagement/Viasoft.Licensing.LicensingManagement.Domain/DTOs/LicensedBundle/LicensedBundleUpdateInput.cs
using System;
using Viasoft.Data.Attributes;
using Viasoft.Licensing.LicensingManagement.Domain.Enums;

namespace Viasoft.Licensing.LicensingManagement.Domain.DTOs.LicensedBundle
{
    public class LicensedBundleUpdateInput
    {
        [StrictRequired]
        public Guid LicensedTenantId { get; set; }
        
        [StrictRequired]
        public Guid BundleId { get; set; }

        [StrictRequired(AllowZeroNumeric = true)]
        public LicensedBundleStatus Status { get; set; }
        
        [StrictRequired(AllowNegativeNumeric = false, AllowZeroNumeric = true)]
        public int NumberOfLicenses { get; set; }
        
        [StrictRequired(AllowNegativeNumeric = false, AllowZeroNumeric = true)]
        public int NumberOfTemporaryLicenses { get; set; }

        public DateTime? ExpirationDateOfTemporaryLicenses { get; set; }
        public DateTime? ExpirationDateTime { get; set; }
        
        [StrictRequired]
        public LicensingModels LicensingModel { get; set; }
        
        public LicensingModes? LicensingMode { get; set; }
    }
}

File: viasoft.licensing.licensingmanagement-frontend/apps/licensing-management/src/app/tokens/inputs/licensed-bundle-update.input.ts
import { LicensedBundleStatus } from '../enum/licensed-bundle-status.enum';
import { LicensingMode } from '../enum/licensing-mode.enum';
import { LicensingModel } from '../enum/licensing-model.enum';

export class LicensedBundleUpdateInput {
    licensedTenantId: string;
    bundleId: string;
    status: LicensedBundleStatus;
    numberOfLicenses: number;
    numberOfTemporaryLicenses: number;
    expirationDateOfTemporaryLicenses?: Date | null;
    licensingModel: LicensingModel;
    licensingMode: LicensingMode;
    expirationDateTime?: Date | null;
}
